### ISSUE TITLE
Mutable sequence in callable parameter is not invariant

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1228]
**Describe the bug**
A function annotated to take an argument of type `T` and a callable with an argument of the same type does not generate an error if `T` is a mutable sequence and the callable is parametrised with a type which is a supertype of the mutable sequence's parameter.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That `['']` would be incompatible with `baz`.

**Screenshots or Code**

```python
# pyright: strict

from typing import Callable, TypeVar

T = TypeVar('T')

def foo(value: T, callback: Callable[[T], None]) -> None:
    ...


def bar(value: str) -> None:
    ...

foo('', bar)  # ok

def baz(value: list[object]) -> None:
    ...

foo([''], baz)  # should be rejected

```

**VS Code extension or command-line**
Either, 1.1.91

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now fixed in Pyright 1.1.92, which I just published. It will also be addressed in next week's Pylance release.

**Comment by layday** [comment]
Thanks for the fix.  However, I believe this issue persists when the type is `None`:

```py
foo([None], baz)
```

It seems to affect all `Generic`s with a type variable which is invariant:

```py
class Qux(Generic[T]):
    ...

def quux(value: Qux[object]) -> None:
    ...

foo(Qux[None](), quux)  # should be rejected

```

**Comment by erictraut** [comment]
Yes, the same bug that I fixed for TypeVars also affected None, which is a special-case type in the type checker. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.93, which I just published. It will also be included in the next release of Pylance.
